File: src/main/java/io/endertech/util/StringHelper.java
package io.endertech.util;

import net.minecraft.util.EnumChatFormatting;
import net.minecraftforge.fluids.FluidStack;
import java.text.DecimalFormat;

public class StringHelper
{
    public static final String holdShiftForDetails = EnumChatFormatting.GRAY + "Hold " + EnumChatFormatting.YELLOW + EnumChatFormatting.ITALIC + "Shift " + EnumChatFormatting.RESET + EnumChatFormatting.GRAY + "for Details" + EnumChatFormatting.RESET;
    public static DecimalFormat twoDP = new DecimalFormat("#.##");

    public static String getEnergyString(int energy)
    {
        if (energy >= 1000000)
        {
            return String.valueOf(twoDP.format(energy / 1000000.0)) + "M";
        } else if (energy >= 1000)
        {
            return String.valueOf(energy / 1000) + "k";
        } else
        {
            return String.valueOf(energy);
        }
    }

    public static String getFluidName(FluidStack fluidStack)
    {
        return cofh.lib.util.helpers.StringHelper.getFluidName(fluidStack);
    }
}

File: src/main/java/io/endertech/util/helper/StringHelper.kt
package io.endertech.util.helper

import net.minecraft.util.EnumChatFormatting
import net.minecraftforge.fluids.FluidStack
import java.text.DecimalFormat

object StringHelper {
    val holdShiftForDetails = EnumChatFormatting.GRAY + LocalisationHelper.localiseString("info.hold_for_details.hold") + " " + EnumChatFormatting.YELLOW + EnumChatFormatting.ITALIC + LocalisationHelper.localiseString("info.hold_for_details.shift") + EnumChatFormatting.RESET + " " + EnumChatFormatting.GRAY + LocalisationHelper.localiseString("info.hold_for_details.for_details") + EnumChatFormatting.RESET
    var twoDP = DecimalFormat("#.##")

    fun getEnergyString(energy: Int): String {
        if (energy == Integer.MAX_VALUE) return LocalisationHelper.localiseString("info.infinite")

        if (energy >= 1000000) {
            return twoDP.format(energy / 1000000.0).toString() + "M"
        } else if (energy >= 1000) {
            return (energy / 1000).toString() + "k"
        } else {
            return energy.toString()
        }
    }

    fun getFluidName(fluidStack: FluidStack): String {
        return cofh.lib.util.helpers.StringHelper.getFluidName(fluidStack)
    }
}
